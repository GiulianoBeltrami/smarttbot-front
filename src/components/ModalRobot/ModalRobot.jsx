import React from 'react'
import { Modal, Button, Row, Col, Container, Form } from 'react-bootstrap';
import { useGetStrategiesQuery } from '../../services/smarttbotApi';
import Loader from '../Loader/Loader';

export const ModalRobot = ({ show, handleClose, onSubmit, data }) => {

    const strategies = data?.data;
    
    return (
        <>
            <Modal backdropClassName="backDropColor" dialogClassName="modalWidth" size="lg" show={show} onHide={handleClose}>
                <Modal.Header className="pb-0" style={{ borderBottom: '0' }} closeButton>
                    <p className="titleText" style={{ fontSize: '16px' }}>Adicionar novo Robô</p>
                </Modal.Header>
                <Modal.Body className="ps-1 py-0">
                    <Container fluid>
                        <Row>
                            <Col>
                                <p className="titleText mb-0" style={{ fontSize: '25px', color: 'background: #5F5D60' }}>Vamos criar seu robô</p>
                                <p className="lightText">Preencha as informações abaixo</p>
                            </Col>
                        </Row>
                        <Row className="pt-3">
                            <Col>
                                <Form>
                                    <Form.Group className="mb-3" controlId="formProductName">
                                        <Form.Label className="formLabelText">Nome do produto</Form.Label>
                                        <Form.Control className="lightText py-2 px-2" placeholder="Nome do produto" />
                                    </Form.Group>

                                    <Form.Group className="mb-3" controlId="formStartCapital">
                                        <Form.Label className="formLabelText">Capital inicial do robô</Form.Label>
                                        <Form.Control className="lightText py-2 px-2" type="number" placeholder="R$" />
                                    </Form.Group>
                                    <Row className="mb-5">
                                        <Col className="col-12">
                                            <p className="titleText" style={{ fontWeight: '500' }}>Selecione uma das estratégias abaixo</p>
                                        </Col>
                                        <Col className="col-12">
                                            <div className="d-grid gap-2">
                                                {strategies.map((element, index) => (
                                                    <Button className="strategyButton text-start p-3" key={index} size="lg">
                                                        {element.name}
                                                    </Button>
                                                ))}
                                            </div>
                                        </Col>
                                    </Row>
                                    <Row className="mb-4">
                                        <Col className="col-6 text-start">
                                            <Button data-testid="closeModal" className="modalCancelButton" onClick={handleClose}>
                                                Cancelar
                                            </Button>
                                        </Col>
                                        <Col className="col-6 text-end">
                                            <Button type="submit" className="modalCreateButton" onClick={onSubmit}>
                                                Criar robô
                                            </Button>
                                        </Col>
                                    </Row>
                                </Form>
                            </Col>
                        </Row>
                    </Container>
                </Modal.Body>
            </Modal>
        </>
    );
}

export default ModalRobot;