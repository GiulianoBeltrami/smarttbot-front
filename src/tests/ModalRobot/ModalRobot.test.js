import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import ModalRobot from '../../components/ModalRobot/ModalRobot';

test('show modal correctly', async () => {

    const handleClose = jest.fn();

    const { container } = render(<ModalRobot show={true} handleClose={handleClose} />);

    expect(container).toMatchSnapshot();

    expect(await screen.findByText('Vamos criar seu robô')).toBeInTheDocument();

});

test('close modal correctly', async () => {

    const handleClose = jest.fn();

    const { container } = render(<ModalRobot show={true} handleClose={handleClose} />);

    expect(container).toMatchSnapshot();

    fireEvent.click(await screen.findByTestId('closeModal'));

    expect(handleClose).toHaveBeenCalledTimes(1);
});

test('render Adicionar novo Robô text', async () => {
    const handleClose = jest.fn();
    const { container } = render(<ModalRobot show={true} handleClose={handleClose} />);
    expect(container).toMatchSnapshot();
    const textElement = await screen.findByText('Adicionar novo Robô');
    expect(textElement).toBeInTheDocument();
})


test('form with valid inputs', async () => {

    const handleClose = jest.fn();
    const handleSubmit = jest.fn();

    const { container } = await render(<ModalRobot show={true} handleClose={handleClose} onSubmit={handleSubmit} />);
    expect(container).toMatchSnapshot();

    fireEvent.change(screen.getByLabelText('Nome do produto'),{target:{value:"Robo1"}});
    fireEvent.change(screen.getByLabelText('Capital inicial do robô'),{target:{value:"250"}});

    fireEvent.click(screen.getByText('Criar robô'))
    expect(handleSubmit).toHaveBeenCalledTimes(1);
})
