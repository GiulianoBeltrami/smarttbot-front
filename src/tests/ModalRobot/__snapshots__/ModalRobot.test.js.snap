// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`close modal correctly 1`] = `<div />`;

exports[`form with valid inputs 1`] = `<div />`;

exports[`render Adicionar novo Rob√¥ text 1`] = `<div />`;

exports[`show modal correctly 1`] = `<div />`;
